compiler/rustc_resolve/src/access_levels.rs

```diff
@@ -123,100 +112,75 @@ impl<'r, 'a> AccessLevelsVisitor<'r, 'a> {
             old_level
         }
     }
+
+    fn get_prev_access_level(&mut self, def_id: LocalDefId) -> Option<AccessLevel> {
+        match self.r.opt_local_parent(def_id) {
```

Comment:
```suggestion
        self.r.local_parent(def_id) {
```
`ast::Item` always has a parent (this function can be inlined and removed).